src/configuration.ts

```diff
@@ -1,16 +1,28 @@
-import { normalizePath } from '../utils';
+import { normalizePath } from './utils';
+
+export interface TargetParseConfiguration {
+  /**
+   * Source file paths. [Glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) are allowed.
+   * If it is a relative path, it will be resolved based on the configuration file path.
+   *
+   * For example, `["src/api/IEditor.ts", "src/bridge/*.ts"]`
+   */
+  source: string[];
+  /**
+   * Interface names for detecting exported modules. If defined, only interfaces that extends the specified interfaces will be parsed.
+   * If not defined, interfaces with JSDoc tag `@shouldExport true` would be parsed
+   */
+  extendedInterfaces?: string[];
```

Comment:
What could be the advantages of extendedInterfaces compared with `@shouldExport`? Could you give some examples? Feels like the string here need to be sync with the code, could be harder.

And also, the name of this property is not straightforward, I have to find the comment here

Reply:
So we don't need to write `@shouldExport` for every interface that needs to be exported.

Yeah, I couldn't find a better name. Any suggestion?

Reply:
> So we don't need to write @shouldExport for every interface that needs to be exported. 

Instead, we need to maintain the string array for the exported interface? There is no restriction between the `extendedInterfaces` array and the actual interface name, so if the interface's name changes how to make sure we also change the name here?

> Yeah, I couldn't find a better name. Any suggestion?
For me the most confusing thing is why it's not `exportedInterfaces`, instead `extendedInterfaces`

Reply:
> Instead, we need to maintain the string array for the exported interface? 

It's not the interfaces listed in this property would be exported, it's the interfaces that extends the interfaces in this property. That's why the property is named `extendedInterfaces`. For example, if we set `extendedInterfaces` to `['Foobar']`, then any interface that extends `'Foobar'` would be exprted.

> There is no restriction between the extendedInterfaces array and the actual interface name, so if the interface's name changes how to make sure we also change the name here?

Unfortunately, there is no way to ensure that. There's not much difference than `@shouldExport` actually. If we forget to add the `@shouldExport` tag to an interface, or if there is a typo, the interface would simply be ignored.

Reply:
I got it. So we can define an interface like `ExportedInterface` and then the interface to be exported could extend it.
What about the name like `ExportedInterfaceBase` or `ExportedProtocol`?

Reply:
Thanks for the suggestion. Renamed.